sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -1460,6 +1468,141 @@ public Mono<Void> rollbackTransaction(ServiceBusTransactionContext transactionCo
+     * @param options options used to delete the messages.
+     * @return a {@link Mono} indicating the number of messages deleted.
+     *
+     * @throws IllegalArgumentException when the {@code messageCount} is less than 1 or exceeds the maximum allowed, as
+     * determined by the Service Bus service.
+     * @throws NullPointerException if {@code options} is null.
+     */
+    public Mono<Integer> deleteMessages(int messageCount, DeleteMessagesOptions options) {
+        return deleteMessages(messageCount, options, receiverOptions.getSessionId());
+    }
+
+    /**
+     * Attempts to purge all messages from an entity.
+     * <p>If the lock for a message is held by a receiver, it will be respected and the message will not be deleted.</p>
+     * <p>
+     * This method may invoke multiple service requests to delete all messages. Because multiple service requests may be
+     * made, the possibility of partial success exists, in such scenario, the method will stop attempting to delete
+     * additional messages and throw the exception that was encountered. Also, due to the multiple service requests,
+     * purge operation may exceed the configured {@link AmqpRetryOptions#getTryTimeout()}.
+     * </p>
+     *
+     * @return a {@link Mono} indicating the number of messages deleted.
+     */
+    public Mono<Integer> purgeMessages() {
+        return purgeMessages(new PurgeMessagesOptions());
+    }
+
+    /**
+     * Attempts to purge all messages from an entity.
+     * <p>If the lock for a message is held by a receiver, it will be respected and the message will not be deleted.</p>
+     * <p>
+     * This method may invoke multiple service requests to delete all messages. Because multiple service requests may be
+     * made, the possibility of partial success exists, in such scenario, the method will stop attempting to delete
+     * additional messages and throw the exception that was encountered. Also, due to the multiple service requests,
+     * purge operation may exceed the configured {@link AmqpRetryOptions#getTryTimeout()}.
+     * </p>
+     *
+     * @param options options used to purge the messages.
+     *
+     * @return a {@link Mono} indicating the number of messages deleted.
+     * @throws NullPointerException if {@code options} is null.
+     */
+    public Mono<Integer> purgeMessages(PurgeMessagesOptions options) {
+        if (Objects.isNull(options)) {
+            return monoError(LOGGER, new NullPointerException("'options' cannot be null."));
+        }
+        final DeleteMessagesOptions deleteMessagesOptions;
+        if (options.getBeforeEnqueueTimeUtc() != null) {
+            deleteMessagesOptions = new DeleteMessagesOptions()
+                .setBeforeEnqueueTimeUtc(options.getBeforeEnqueueTimeUtc());
+        } else {
+            deleteMessagesOptions = new DeleteMessagesOptions()
+                .setBeforeEnqueueTimeUtc(OffsetDateTime.now());
```

Comment:
AFAIK, `offsetdatetime.now()` is the current system datetime? Would we want to use UTC instead?

Reply:
thank you!!, awesome catch, will fix this.